diff -Naur ./redhat/babeld.service ./redhat/babeld.service
--- ./redhat/babeld.service	1970-01-01 10:00:00.000000000 +1000
+++ ./redhat/babeld.service	2016-10-17 13:27:25.925457328 +1000
@@ -0,0 +1,15 @@
+[Unit]
+Description=Babel routing daemon
+BindsTo=zebra.service
+After=zebra.service
+ConditionPathExists=/etc/quagga/babeld.conf
+
+[Service]
+Type=forking
+PIDFile=/run/quagga/babeld.pid
+EnvironmentFile=/etc/sysconfig/quagga
+ExecStart=/usr/sbin/babeld -d $OSPF6D_OPTS -f /etc/quagga/babeld.conf
+Restart=on-abort
+
+[Install]
+WantedBy=multi-user.target
diff -Naur ./redhat/bgpd.service ./redhat/bgpd.service
--- ./redhat/bgpd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/bgpd.service	2016-10-17 13:12:50.158617446 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=BGP routing daemon
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/bgpd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/bgpd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/bgpd -d $BGPD_OPTS -f /etc/quagga/bgpd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/isisd.service ./redhat/isisd.service
--- ./redhat/isisd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/isisd.service	2016-10-17 13:13:22.526500649 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=IS-IS routing daemon
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/isisd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/isisd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/isisd -d $ISISD_OPTS -f /etc/quagga/isisd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/ospf6d.service ./redhat/ospf6d.service
--- ./redhat/ospf6d.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/ospf6d.service	2016-10-17 13:13:57.702373719 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=OSPF routing daemon for IPv6
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/ospf6d.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/ospf6d.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/ospf6d -d $OSPF6D_OPTS -f /etc/quagga/ospf6d.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/ospfd.service ./redhat/ospfd.service
--- ./redhat/ospfd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/ospfd.service	2016-10-17 13:14:27.374266652 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=OSPF routing daemon
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/ospfd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/ospfd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/ospfd -d $OSPFD_OPTS -f /etc/quagga/ospfd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/pimd.service ./redhat/pimd.service
--- ./redhat/pimd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/pimd.service	2016-10-17 13:17:23.150632381 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=PIM multicast routing engine
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/pimd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/pimd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/pimd -d $PIMD_OPTS -f /etc/quagga/pimd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/ripd.service ./redhat/ripd.service
--- ./redhat/ripd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/ripd.service	2016-10-17 13:15:01.230144486 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=RIP routing daemon
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/ripd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/ripd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/ripd -d $RIPD_OPTS -f /etc/quagga/ripd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/ripngd.service ./redhat/ripngd.service
--- ./redhat/ripngd.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/ripngd.service	2016-10-17 13:15:36.599016861 +1000
@@ -1,14 +1,15 @@
 [Unit]
 Description=RIP routing daemon for IPv6
-BindTo=zebra.service
-After=syslog.target network.target zebra.service
+BindsTo=zebra.service
+After=zebra.service
 ConditionPathExists=/etc/quagga/ripngd.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/ripngd.pid
 EnvironmentFile=/etc/sysconfig/quagga
 ExecStart=/usr/sbin/ripngd -d $RIPNGD_OPTS -f /etc/quagga/ripngd.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
diff -Naur ./redhat/zebra.service ./redhat/zebra.service
--- ./redhat/zebra.service	2016-10-17 13:08:33.866542249 +1000
+++ ./redhat/zebra.service	2016-10-17 13:16:08.958900093 +1000
@@ -1,14 +1,16 @@
 [Unit]
 Description=GNU Zebra routing manager
-After=syslog.target network.target
+Wants=network.target
+Before=network.target
 ConditionPathExists=/etc/quagga/zebra.conf
 
 [Service]
 Type=forking
+PIDFile=/run/quagga/zebra.pid
 EnvironmentFile=-/etc/sysconfig/quagga
 ExecStartPre=/sbin/ip route flush proto zebra
 ExecStart=/usr/sbin/zebra -d $ZEBRA_OPTS -f /etc/quagga/zebra.conf
 Restart=on-abort
 
 [Install]
-WantedBy=network.target
+WantedBy=multi-user.target
